support-files/mariadb.logrotate.in

```diff
@@ -0,0 +1,54 @@
+# This is the MariaDB configuration for the logrotate utility
+#
+# Note that on most Linux systems logs are written to journald, which has its
+# own rotation scheme.
+#
+# Read https://mariadb.com/kb/en/error-log/ to learn more about logging and
+# https://mariadb.com/kb/en/rotating-logs-on-unix-and-linux/ about rotating logs.
+
+@localstatedir@/mysqld.log @logdir@/mysql.log @localstatedir@/mariadb.log @logdir@/mysql-slow.log @logdir@/mariadb-slow.log @logdir@/error.log {
+
+  # If any of the files listed above is missing, skip them silently without
+  # emitting any errors
+  missingok
+
+  # If file exists but is empty, don't rotate it
+  notifempty
+
+  # Run daily
+  daily
+
+  # Keep one week of logs
+  rotate 7
+
+  # If file is growing too big, rotate immediately
+  maxsize 100M
+
+  # Compress logs, as they are text and compression will save a lot of disk space
+  compress
+
+  # Don't compress the log immediately to avoid errors about "file size changed while zipping"
+  delaycompress
+
+  # Don't run the postrotate script for each file configured in this file, but
+  # run it only once if one or more files were rotated
+  sharedscripts
+
+  # After each rotation, run this custom script to flush the logs. Note that
+  # this assumes that the mariadb-admin command has database access, which it
+  # has thanks to the default use of Unix socket authentication for the 'root'
```

Comment:
I suspect rpm based uses the `mysql` user -  support-files/rpm/server-postin.sh + scripts/mysql_install_db.sh which means they'll need a 'user mysql' in this file.

Reply:
Why don't we have Unix socket authentication everywhere? If user 'mysql' is needed in rpm world, the postrorate -x could be extended with another test, maybe check for user mysql in /etc/passwd and then prepend the mariadb-admin call with `su mysql` or something?

Reply:
It is unix socket auth, its just using the 'mysql' user. It just needs to be checked that this implementation falls under the permitted selinux rules.

Reply:
User 'mysql' is not here now has this been tested for SELinux. Neither was in the legacy logrotate file this replaces, so it should not be a step backward.

I suggest you merge this and do additional changes in a follow-up commit.